File: Android/SimploC/app/src/main/java/com/localhost/lin/simploc/Fragments/CreditFragment.java
package com.localhost.lin.simploc.Fragments;

import android.content.Context;
import android.net.Uri;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ListView;
import android.widget.SimpleAdapter;

import com.localhost.lin.simploc.R;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.Iterator;
import java.util.List;
import java.util.Map;

/**
 * A simple {@link Fragment} subclass.
 * Activities that contain this fragment must implement the
 * {@link CreditFragment.OnFragmentInteractionListener} interface
 * to handle interaction events.
 * Use the {@link CreditFragment#newInstance} factory method to
 * create an instance of this fragment.
 */
public class CreditFragment extends Fragment {
    // TODO: Rename parameter arguments, choose names that match
    // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
    private static final String ARG_PARAM1 = "param";

    // TODO: Rename and change types of parameters
    private String mParam;
    private static String[] mColumns = new String[]{"class", "need", "get", "nopass", "rest"};

    private OnFragmentInteractionListener mListener;

    public CreditFragment() {
        // Required empty public constructor
    }

    /**
     * Use this factory method to create a new instance of
     * this fragment using the provided parameters.
     *
     * @param param Parameter 1.
     * @return A new instance of fragment CreditFragment.
     */
    // TODO: Rename and change types and number of parameters
    public static CreditFragment newInstance(String param) {
        CreditFragment fragment = new CreditFragment();
        Bundle args = new Bundle();
        args.putString(ARG_PARAM1, param);
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        if (getArguments() != null) {
            mParam = getArguments().getString(ARG_PARAM1);
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View view = inflater.inflate(R.layout.fragment_credit2, container, false);
        ListView listView = (ListView)view.findViewById(R.id.lv_credit);
        List<HashMap<String, String> > creditData = getJsonData(mParam);
        SimpleAdapter adapter = new SimpleAdapter(view.getContext(), creditData,
                R.layout.list_item_credit, mColumns,
                new int[]{R.id.credit_name, R.id.credit_need, R.id.credit_get, R.id.credit_onpass, R.id.credit_rest});
        listView.setAdapter(adapter);
        return view;
    }

    // TODO: Rename method, update argument and hook method into UI event
    public void onButtonPressed(Uri uri) {
        if (mListener != null) {
            mListener.onFragmentInteraction(uri);
        }
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof OnFragmentInteractionListener) {
            mListener = (OnFragmentInteractionListener) context;
        } else {
            throw new RuntimeException(context.toString()
                    + " must implement OnFragmentInteractionListener");
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mListener = null;
    }

    private List<HashMap<String, String> > getJsonData(String data) {
        List<HashMap<String, String> > result = new ArrayList<>();
        try {
            JSONArray credits = new JSONArray(data);
            for(int i=0; i<credits.length(); i++) {
                JSONObject jsonItem = credits.getJSONObject(i);
                HashMap<String, String> item = new HashMap<>();
                Iterator<String> keys = jsonItem.keys();
                while (keys.hasNext()) {
                    String key = keys.next();
                    item.put(key, jsonItem.getString(key));
                }
                result.add(item);
            }
        } catch (JSONException e) {
            e.printStackTrace();
        }
        return result;
    }

    /**
     * This interface must be implemented by activities that contain this
     * fragment to allow an interaction in this fragment to be communicated
     * to the activity and potentially other fragments contained in that
     * activity.
     * <p/>
     * See the Android Training lesson <a href=
     * "http://developer.android.com/training/basics/fragments/communicating.html"
     * >Communicating with Other Fragments</a> for more information.
     */
    public interface OnFragmentInteractionListener {
        // TODO: Update argument type and name
        void onFragmentInteraction(Uri uri);
    }
}

File: Android/SimploC/app/src/main/java/com/localhost/lin/simploc/Fragments/CreditFragment.kt
package com.localhost.lin.simploc.Fragments

import android.content.Context
import android.net.Uri
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ListView
import android.widget.SimpleAdapter
import com.localhost.lin.simploc.R
import org.json.JSONArray
import org.json.JSONException
import java.util.*

/**
 * A simple [Fragment] subclass.
 * Activities that contain this fragment must implement the
 * [CreditFragment.OnFragmentInteractionListener] interface
 * to handle interaction events.
 * Use the [CreditFragment.newInstance] factory method to
 * create an instance of this fragment.
 */
class CreditFragment : Fragment() {

    // TODO: Rename and change types of parameters
    private var mParam: String? = null

    private var mListener: OnFragmentInteractionListener? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        if (arguments != null) {
            mParam = arguments.getString(ARG_PARAM1)
        }
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        // Inflate the layout for this fragment
        val view = inflater!!.inflate(R.layout.fragment_credit2, container, false)
        val listView = view.findViewById(R.id.lv_credit) as ListView
        val creditData = getJsonData(mParam.toString())
        val adapter = SimpleAdapter(view.context, creditData,
                R.layout.list_item_credit, mColumns,
                intArrayOf(R.id.credit_name, R.id.credit_need, R.id.credit_get, R.id.credit_onpass, R.id.credit_rest))
        listView.adapter = adapter
        return view
    }

    // TODO: Rename method, update argument and hook method into UI event
    fun onButtonPressed(uri: Uri) {
        if (mListener != null) {
            mListener!!.onFragmentInteraction(uri)
        }
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        if (context is OnFragmentInteractionListener) {
            mListener = context as OnFragmentInteractionListener?
        } else {
            throw RuntimeException(context!!.toString() + " must implement OnFragmentInteractionListener")
        }
    }

    override fun onDetach() {
        super.onDetach()
        mListener = null
    }

    private fun getJsonData(data: String): List<HashMap<String, String>> {
        val result = ArrayList<HashMap<String, String>>()
        try {
            val credits = JSONArray(data)
            for (i in 0..credits.length() - 1) {
                val jsonItem = credits.getJSONObject(i)
                val item = HashMap<String, String>()
                val keys = jsonItem.keys()
                while (keys.hasNext()) {
                    val key = keys.next()
                    item.put(key, jsonItem.getString(key))
                }
                result.add(item)
            }
        } catch (e: JSONException) {
            e.printStackTrace()
        }

        return result
    }

    /**
     * This interface must be implemented by activities that contain this
     * fragment to allow an interaction in this fragment to be communicated
     * to the activity and potentially other fragments contained in that
     * activity.
     *
     *
     * See the Android Training lesson [Communicating with Other Fragments](http://developer.android.com/training/basics/fragments/communicating.html) for more information.
     */
    interface OnFragmentInteractionListener {
        // TODO: Update argument type and name
        fun onFragmentInteraction(uri: Uri)
    }

    companion object {
        // TODO: Rename parameter arguments, choose names that match
        // the fragment initialization parameters, e.g. ARG_ITEM_NUMBER
        private val ARG_PARAM1 = "param"
        private val mColumns = arrayOf("class", "need", "get", "nopass", "rest")

        /**
         * Use this factory method to create a new instance of
         * this fragment using the provided parameters.

         * @param param Parameter 1.
         * *
         * @return A new instance of fragment CreditFragment.
         */
        // TODO: Rename and change types and number of parameters
        fun newInstance(param: String): CreditFragment {
            val fragment = CreditFragment()
            val args = Bundle()
            args.putString(ARG_PARAM1, param)
            fragment.arguments = args
            return fragment
        }
    }
}// Required empty public constructor
